api/window/src/main/java/quarks/window/Window.java

```diff
@@ -157,4 +158,10 @@ Licensed to the Apache Software Foundation (ASF) under one
      */
     Function<T, K> getKeyFunction();
 
+    /**
+     * Retrieves the partitions in the window.
+     * @return A map of the window's keys and partitions.
```

Comment:
How about adding doc for the behavior / use requirements of the returned map as they relate to: is the returned map/partitions "stable" or can the state change? What happens if one attempts to change the map or the partitions it contains - is it backed by the live window's objects?